spec.html

```diff
@@ -7875,6 +7880,180 @@ <h1>Forward Progress</h1>
+      </p>
+      <emu-note>
+        WeakRef-obliviousness, together with liveness, capture two notions. One, that a WeakRef itself does not keep an object alive. Two, that cycles in liveness does not imply that an object is live. To be concrete, if determining _obj_'s liveness depends on determining the liveness of another WeakRef referent, _obj2_, _obj2_'s liveness cannot assume _obj_'s liveness, which would beg the question.
+      </emu-note>
+      <emu-note>
+        WeakRef-obliviousness is defined on sets of objects instead of individual objects to account for cycles. If it were defined on individual objects, then an object in a cycle will be considered live even though its Object value is only observed via WeakRefs of other objects in the cycle.
+      </emu-note>
+
+      <p>
+        At any point during evaluation, a set of objects _S_ is considered <dfn>live</dfn> if either of the following conditions is met:
+      </p>
+
+      <ul>
+        <li>
+          Any element in _S_ is included in any agent's [[KeptAlive]] List.
+        </li>
+        <li>
+          There exists a valid future hypothetical WeakRef-oblivious execution with respect to _S_ that observes the Object value of any object in _S_.
+        </li>
+      </ul>
+      <emu-note>
+        The intuition the second condition above intends to capture is that an object is live if its identity is observable via non-WeakRef means. An object's identity may be observed by observing a strict equality comparison between objects or observing the object being used as key in a Map.
+      </emu-note>
+      <emu-note>
+        <p>
+        Presence of an object in a field, an internal slot, or a property does not imply that the object is live. For example if the object in question is never passed back to the program, then it cannot be observed.
+        </p>
+
+        <p>
+        This is the case for keys in a WeakMap, members of a WeakSet, as well as the [[WeakRefTarget]] and [[UnregisterToken]] fields of a FinalizationRegistry Cell record.
+        </p>
+
+        <p>
+        The above definition implies that, if a key in a WeakMap is not live, then its corresponding value is not necessarily live either.
+        </p>
+      </emu-note>
+      <emu-note>
+        Liveness is the lower bound for guaranteeing which WeakRefs that engines must not empty. In practice, liveness as defined here is undecidable and engines use conservative approximations such as reachability. There is expected to be significant implementation leeway.
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-weakref-execution">
+      <h1>Execution</h1>
+
+      <p>At any time, if a set of objects _S_ is not live, an ECMAScript implementation may perform the following steps atomically:</p>
```

Comment:
My general thought was to use "Object or Symbol" all over the place. If we use the HasIdentity operation as a chosen name, we could say, "values with identity" all over the place instead. In any case, I think this should be a pretty mechanical find-and-replace with whatever wording we decide on.

Reply:
I will wait for a resolution here before changing this.

Reply:
I don't expect adding new kinds of values with identity in the near future, and since there are only two, I'm leaning towards "Object or Symbol" over "values with identity", since "values with identity" is also 3 words and would need to be enumerated in another section that the reader clicks through to find what are the values with identity anyway.

Reply:
the primary advantage i see to coming up with a single term for it is that there'd be a linkable dfn which serves as the single source of truth for that that means. perhaps since it'll only ever be two items, it's not that important.

Reply:
Can we move this discussion to the [Symbols as WeakMap keys repo](https://github.com/tc39/proposal-symbols-as-weakmap-keys)? This PR will not be one where we land any wording on this question. I encourage people who have opinions here to file an issue. I hope that this PR will land before that proposal goes to Stage 3, and so we have plenty of time to work out the wording.